targets/arm/mikroe/sam/include/gpio/implementations/implementation_1/hal_ll_gpio_struct_type.h

```diff
@@ -0,0 +1,107 @@
+/****************************************************************************
+**
+** Copyright (C) ${COPYRIGHT_YEAR} MikroElektronika d.o.o.
+** Contact: https://www.mikroe.com/contact
+**
+** This file is part of the mikroSDK package
+**
+** Commercial License Usage
+**
+** Licensees holding valid commercial NECTO compilers AI licenses may use this
+** file in accordance with the commercial license agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and The MikroElektronika Company.
+** For licensing terms and conditions see
+** https://www.mikroe.com/legal/software-license-agreement.
+** For further information use the contact form at
+** https://www.mikroe.com/contact.
+**
+**
+** GNU Lesser General Public License Usage
+**
+** Alternatively, this file may be used for
+** non-commercial projects under the terms of the GNU Lesser
+** General Public License version 3 as published by the Free Software
+** Foundation: https://www.gnu.org/licenses/lgpl-3.0.html.
+**
+** The above copyright notice and this permission notice shall be
+** included in all copies or substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** OF MERCHANTABILITY, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+** TO THE WARRANTIES FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+** IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+** DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT
+** OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
+** OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+**
+****************************************************************************/
+/*!
+ * @file  hal_ll_gpio_struct_type.h
+ * @brief GPIO HAL LL register structure typedef.
+ */
+
+#ifndef _HAL_LL_GPIO_STRUCT_TYPE_H_
+#define _HAL_LL_GPIO_STRUCT_TYPE_H_
+
+#ifdef __cplusplus
+extern "C"{
+#endif
+
+#include <stdint.h>
+
+/**
+ *   GPIO module interrupt registers control structure
+ **/
+typedef struct hal_ll_gpio_interrupt_register_handle
+{
+    uint32_t enable;
```

Comment:
Can we use `hal_ll_base_addr_t` typedef instead of `uint32_t`?

Reply:
For all implementations including the AI-generated ones we use uint32_t for gpio headers. I can make an exception for this file, but I think we can follow the original approach.

Reply:
It should be `hal_ll_base_addr_t` indeed. It is typedefed for a reason. Change it.

Reply:
Changed it for all implementations except GPIO as it doesn't include necessary header files for all architectures.